src/SDPoptimize.jl

```diff
@@ -348,108 +185,78 @@ Returns :
     of the system at each time step
 
 """
-function sdp_solve_HD(model::StochDynProgModel,
-                  param::SDPparameters,
-                  display=0::Int64)
+function sdp_compute_value_functions(model::StochDynProgModel,
+                                        param::SDPparameters,
+                                        display=0::Int64)
 
     TF = model.stageNumber
     next_state = zeros(Float64, model.dimStates)
-    law = model.noises
 
     u_bounds = model.ulim
     x_bounds = model.xlim
     x_steps = param.stateSteps
+    x_dim = model.dimStates
+
+    dynamics = model.dynamics
+    constraints = model.constraints
+    cost = model.costFunctions
+
+    law = model.noises
 
     #Compute cartesian product spaces
     product_states, product_controls = generate_grid(model, param)
 
+    product_states = collect(product_states)
     product_controls = collect(product_controls)
 
-    V = zeros(Float64, param.stateVariablesSizes..., TF)
+    V = SharedArray{Float64}(zeros(Float64, param.stateVariablesSizes..., TF))
 
     #Compute final value functions
     for x in product_states
-        ind_x = index_from_variable(x, x_bounds, x_steps)
+        ind_x = SDPutils.index_from_variable(x, x_bounds, x_steps)
         V[ind_x..., TF] = model.finalCostFunction(x)
     end
 
+    if param.expectation_computation!="MonteCarlo" && param.expectation_computation!="Exact"
+        warn("param.expectation_computation should be 'MonteCarlo' or 'Exact'. Defaulted to 'exact'")
```

Comment:
it is not an error so I didn't put @fact_throw but I did it for the information structure because we can't really choose a default value it could lead to bad results not absurd enough to notice the mistake